pilot/pkg/bootstrap/server.go

```diff
@@ -1052,9 +1089,59 @@ func (s *Server) createPeerCertVerifier(tlsOptions TLSOptions) (*spiffe.PeerCert
 		peerCertVerifier.AddMappings(certMap)
 	}
 
+	if features.PilotCertProvider == constants.CertProviderSocket {
+		caBundle, err := s.x509Source.GetCaBundleForTrustDomain(spiffe.GetTrustDomain())
+		if err != nil {
+			return nil, fmt.Errorf("error gettting CA Bundle for trust domain %v: %v", spiffe.GetTrustDomain(), err)
+		}
+
+		peerCertVerifier.AddMapping(spiffe.GetTrustDomain(), caBundle.X509Authorities())
+
+		// start watcher to keep receiving X.509 bundle updates and add them to the peerCertVerifier
+		s.addStartFunc(func(stop <-chan struct{}) error {
+			go func() {
+				ctx, cancel := context.WithCancel(context.Background())
+				defer cancel()
+				watcher := bundleWatcher{peerCertVerifier}
+				err := spiffe.WatchX509Bundles(ctx, security.WorkloadIdentitySocketPath, watcher)
+				if err != nil {
+					log.Errorf("error starting X.509 Bundle watcher for Pilot: %v", err)
+				}
+				<-stop
+			}()
+			return nil
+		})
+	}
+
 	return peerCertVerifier, nil
 }
 
+// implements workloadapi.X509BundleWatcher interface.
+type bundleWatcher struct {
+	peerCertVerifier *spiffe.PeerCertVerifier
+}
+
+// OnX509BundlesUpdate gets the X.509 bundle updates and adds them to the peerCertVerifier.
+func (b bundleWatcher) OnX509BundlesUpdate(s *x509bundle.Set) {
+	td, err := spiffeid.TrustDomainFromString(spiffe.GetTrustDomain())
+	if err != nil {
+		log.Errorf("error trying to parse trust domain %q reason: %v", td, err)
```

Comment:
is it safe to continue after logging? `td` is being used in the line below

Reply:
No, it should return.